### ISSUE TITLE
bitwuzla_set_termination_callback: wrong behavior or documentation

### ISSUE THREAD

**Issue by recoules** [bitwuzla/bitwuzla#6]
The documentation states that `The callback function, returns a value > 0 if bitwuzla has been terminated.` but it seems that returning a negative value still terminate the solving.

*Note.* I would have understood if the type was unsigned but the signature states `int32_t`.

Here is an example that illustrates the problem (using `CaDiCal`):  
 it is a hard solving problem `a * b = p` that should take a very long time but it directly returns `unknown` while the termination callback always returns `-1`. 
```c++
#include <stdio.h>
#include <bitwuzla.h>

int32_t minus_one(void *state)
{
  return -1;
}

int main (int argc, char *argv[])
{
  Bitwuzla *bitwuzla = bitwuzla_new();
  bitwuzla_set_option(bitwuzla, BITWUZLA_OPT_PRODUCE_MODELS, 2);
  bitwuzla_set_option(bitwuzla, BITWUZLA_OPT_INCREMENTAL, 1);

  const BitwuzlaSort *bv64 = bitwuzla_mk_bv_sort(bitwuzla, 64);
  const BitwuzlaSort *bv128 = bitwuzla_mk_bv_sort(bitwuzla, 128);

  const BitwuzlaTerm *a = bitwuzla_mk_const(bitwuzla, bv64, "a");
  const BitwuzlaTerm *b = bitwuzla_mk_const(bitwuzla, bv64, "b");

  const BitwuzlaTerm *a128 =
    bitwuzla_mk_term1_indexed1(bitwuzla,
			       BITWUZLA_KIND_BV_ZERO_EXTEND,
			       a, 64);
  const BitwuzlaTerm *b128 =
    bitwuzla_mk_term1_indexed1(bitwuzla,
			       BITWUZLA_KIND_BV_ZERO_EXTEND,
			       b, 64);

  const BitwuzlaTerm *ab128 =
    bitwuzla_mk_term2(bitwuzla,
		      BITWUZLA_KIND_BV_MUL,
		      a128, b128);

  const BitwuzlaTerm *p =
    bitwuzla_mk_bv_value(bitwuzla, bv128,
			 "87e03acc9f5050086f083d2d5d6b9d47",
			 BITWUZLA_BV_BASE_HEX);

  const BitwuzlaTerm *o128 =
    bitwuzla_mk_bv_one(bitwuzla, bv128);

  bitwuzla_assert(bitwuzla, bitwuzla_mk_term2(bitwuzla,
					      BITWUZLA_KIND_DISTINCT,
					      a128, o128));
  bitwuzla_assert(bitwuzla, bitwuzla_mk_term2(bitwuzla,
					      BITWUZLA_KIND_DISTINCT,
					      b128, o128));
  bitwuzla_assert(bitwuzla, bitwuzla_mk_term2(bitwuzla,
					      BITWUZLA_KIND_EQUAL,
					      ab128, p));

  bitwuzla_set_termination_callback(bitwuzla, &minus_one, NULL);

  switch (bitwuzla_check_sat(bitwuzla)) {
  case BITWUZLA_SAT:
    printf("sat\n");
    break;
  case BITWUZLA_UNSAT:
    printf("unsat\n");
    break;
  case BITWUZLA_UNKNOWN:
    printf("unknown\n");
    break;
  }

  bitwuzla_delete(bitwuzla);
  return 0;
}
```

My workaround is currently to return `0` also for negative values:
```c
int32_t x = termination_callback(state);
return x & ~(x >> (8 * sizeof(int32_t) - 1));
```

**Comment by mpreiner** [comment]
Yes, the documentation is wrong here. It should actually be != 0 instead of > 0. Fixed with 76865c90.

**Comment by recoules** [comment]
Thanks.
